### ISSUE TITLE
Checkpointing Features

### ISSUE THREAD

**Issue by bbalaji-ucsd** [NervanaSystems/coach#177]
Right now Coach saves checkpoints every X seconds. It would be great if I can save checkpoints every X iterations or save checkpoints if it reaches an evaluation reward threshold. 

It will also help if I can save checkpoints as a protobuf file, so I can use the saved model independent of Coach.

**Comment by safrooze** [comment]
So you want to save a checkpoint at a certain reward threshold but continue training? Is this in case you notice overfit later on and can go back to an earlier set of parameters?

**Comment by bbalaji-ucsd** [comment]
Yes, to both questions. Although flexibility to stop training after a reward threshold also helps. In the latter case, it would help to save a checkpoint at the end of training.

**Comment by safrooze** [comment]
There already is ability to stop training when a specific reward is achieved: "`-asc`" or "`--apply_stop_condition`" will stop early if a specific reward is hit and saves a checkpoint at the end.

**Comment by bbalaji-ucsd** [comment]
Ok, that solves one of the problems then :)
